litellm/utils.py

```diff
@@ -12419,14 +12419,12 @@ def shorten_message_to_fit_limit(message, tokens_needed, model):
         return message
 
     content = message["content"]
+    total_tokens = get_token_count([message], model)
+    prev_tokens = total_tokens + 1
 
-    while True:
-        total_tokens = get_token_count([message], model)
-
-        if total_tokens <= tokens_needed:
-            break
+    while prev_tokens > total_tokens > tokens_needed:
```

Comment:
what is happening here?

Reply:
The actual condition is `total_tokens < prev_tokens and total_tokens > token_needed`  , i.e. only continue the loop if the current token is less than the previous token (this ensure some trimming is happening) and current token is above token needed ( the base case to trim). PyCharm have chained the expression.